src/mlpack/core/util/mlpack_main.hpp

```diff
@@ -154,10 +154,11 @@ using Option = mlpack::bindings::tests::TestOption<T>;
 #include <mlpack/core/util/param.hpp>
 
 #undef PROGRAM_INFO
-#define PROGRAM_INFO(NAME, SHORT_DESC, DESC, ...) \
-    static mlpack::util::ProgramDoc \
-    cli_programdoc_dummy_object = mlpack::util::ProgramDoc(NAME, SHORT_DESC, \
-    []() { return DESC; }, { __VA_ARGS__ })
+#define PROGRAM_INFO(NAME, SHORT_DESC, DESC, EXAMPLE, ...) static \
+    mlpack::util::ProgramDoc \
+    io_programdoc_dummy_object = mlpack::util::ProgramDoc(NAME, SHORT_DESC, \
+    []() { return std::string(DESC) + std::string(EXAMPLE); }, []() \
+    { return ""; }, { __VA_ARGS__ })
```

Comment:
Nice, this is one option for what I was thinking.  A couple further thoughts:

 - It looks like this only allows one `EXAMPLE`, so we don't have a way to split into multiple examples.
 - The number of parameters to `PROGRAM_INFO` is getting large.  What would you think about splitting `PROGRAM_INFO` into a bunch of macros, like this:

```
BINDING_NAME("program name");
BINDING_SHORT_DESC("This is a short, two-sentence description of what the program does.");
BINDING_LONG_DESC("This is a long description of what the program does."
    " It might be many lines long and have lots of details about different options.");
BINDING_EXAMPLE("This contains one example for this particular binding.\n" +
    PROGRAM_CALL(...));
BINDING_EXAMPLE("This contains another example for this particular binding.\n" +
    PROGRAM_CALL(...));
// There could be many of these "see alsos".
BINDING_SEE_ALSO("https://en.wikipedia.org/wiki/Machine_learning");
```

It would take a little bit of refactoring but I wonder if that is a cleaner approach overall.  What you've done in this PR is fine for now, I think, but I do think maybe it's worth some discussion about a longer-term cleanup.  If you think either of those ideas are useful, and we can work it into a quick proposal, we should open an issue back in the main mlpack repository and then we can do it (I can help with the refactoring of course! :)).

Reply:
This looks awesome. We should definitely discuss on this, it will give a great help if you can open a issue in the main mlpack repository. May be it will be better if we handle this in some different PR in the main mlpack repository, as this translation is not directly related to R-bindings. It seems like according to the current progress, I might complete R-bindings before time and this translation may result into a great stretch goal to my timeline.

Reply:
I agree with that---let me open a PR in the mlpack repository later today. :+1:

Reply:
https://github.com/mlpack/mlpack/issues/2521

Reply:
Awesome thank you for opening the issue.